src/service/timer-impl.js

```diff
@@ -78,13 +81,44 @@ export class Timer {
     }
     const wrapped = () => {
       try {
+        if (this.canceled_[id]) {
+          delete this.canceled_[id];
+          return;
+        }
         callback();
       } catch (e) {
         reportError(e);
         throw e;
       }
     };
-    return this.win.setTimeout(wrapped, opt_delay);
+
+    const delayEnd = Date.now() + opt_delay;
+    if (!this.delayQueue_[delayEnd]) {
+      this.delayQueue_[delayEnd] = [];
+      id = this.win.setTimeout(
+        this.callCallbacksForDelay_.bind(this),
+        opt_delay,
+        delayEnd
+      );
+    }
+
+    this.delayQueue_[delayEnd].push(wrapped);
+    return id;
+  }
+
+  /**
+   * Calls previously batched callbacks for a given ms key.
+   * @param {number} delay
+   * @private
+   */
+  callCallbacksForDelay_(delay) {
+    const callbacks = this.delayQueue_[delay];
+    delete this.delayQueue_[delay];
+
+    for (let i = 0; i < callbacks.length; i++) {
+      const cb = callbacks[i];
```

Comment:
This needs to be wrapped in a try-catch, since any `cb` could throw.

Reply:
This cb is already wrapped in a try-catch before being pushed to the queue, or am I missing something?